logicalplan/plan_test.go

```diff
@@ -83,9 +93,49 @@ func TestDefaultOptimizers(t *testing.T) {
 			testutil.Ok(t, err)
 
 			plan := New(expr, time.Unix(0, 0), time.Unix(0, 0))
-			optimizedPlan := plan.Optimize(DefaultOptimizers)
+			initial := plan.Expr().String()
+			plan.Optimize(DefaultOptimizers...)
+			testutil.Equals(t, initial, plan.PreOptimizationExpr().String())
+
 			expectedPlan := strings.Trim(spaces.ReplaceAllString(tcase.expected, " "), " ")
-			testutil.Equals(t, expectedPlan, optimizedPlan.Expr().String())
+			testutil.Equals(t, expectedPlan, plan.Expr().String())
+		})
+	}
+}
+
+func TestDefaultOptimizers_OptimizationApplied(t *testing.T) {
+	cases := []struct {
+		name               string
+		expr               string
+		expectedOptApplied []string
+	}{
+		{
+			name: "common selectors with one matching all",
+			expr: `sum(metric{c="d"}) / sum(metric{})`,
+			expectedOptApplied: []string{
```

Comment:
Maybe we should use a more stable comparison here, something like an optimizer ID. These tests look quite brittle, every time we change the logging the will start to fail.